command/jsonconfig/config.go

```diff
@@ -46,6 +46,7 @@ type moduleCall struct {
 	CountExpression   *expression            `json:"count_expression,omitempty"`
 	ForEachExpression *expression            `json:"for_each_expression,omitempty"`
 	Module            module                 `json:"module,omitempty"`
+	ModuleVersion     string                 `json:"module_version,omitempty"`
```

Comment:
I think there's a potential confusion here between constraints and resolved versions, so perhaps we can call this `version_constraint` since it looks like it'll give e.g. `~> 1.0` rather than the specific version (e.g. `1.0.3`) that was selected matching that constraint.

We may also want to exposed `ResolvedVersion` _somewhere_, but I'm not sure this is the best place for it.

Saying that though, it looks like the code below is populating `ResolvedSource` with the _given_ source, rather than the resolved one (e.g. turning a registry source into the real source it refers to), so it seems like we're already a little confused in that regard.

So, here's what I'd suggest:

* Rename `ResolvedSource` to just `Source`, and keep its current implementation that just writes in exactly what the user specified, rather than any normalized/resolved version of it.
* Rename `ModuleVersion` to `VersionConstraint`, reflecting that it's the constraint string given in config rather than a single version.
* In future we may add back `ResolvedSource` and `ResolvedVersion` to give the normalized source address and the specific selected version respectively, but we don't need to do that right now.

Reply:
Ah, shoot, I Chris also asked about source vs resolved source - I think I just let them bleed together in my head. 
@vancluever, is "source" (as it's working currently) sufficient for your needs?

Reply:
@apparentlymart has it all right here - this is what we need for the Sentinel shim (as these are the values that currently exist when parsing Terraform config). I'm fine with changing the naming. The other values can be added at a later time.